bitnami/seaweedfs/templates/master/service.yaml

```diff
@@ -0,0 +1,78 @@
+{{- /*
+Copyright VMware, Inc.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "seaweedfs.master.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: master
+  {{- if or .Values.master.service.annotations .Values.commonAnnotations }}
+  {{- $annotations := include "common.tplvalues.merge" (dict "values" (list .Values.master.service.annotations .Values.commonAnnotations) "context" .) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  type: {{ .Values.master.service.type }}
+  {{- if and .Values.master.service.clusterIP (eq .Values.master.service.type "ClusterIP") }}
+  clusterIP: {{ .Values.master.service.clusterIP }}
+  {{- end }}
+  {{- if .Values.master.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.master.service.sessionAffinity }}
+  {{- end }}
+  {{- if .Values.master.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.master.service.sessionAffinityConfig "context" $) | nindent 4 }}
+  {{- end }}
+  {{- if or (eq .Values.master.service.type "LoadBalancer") (eq .Values.master.service.type "NodePort") }}
+  externalTrafficPolicy: {{ .Values.master.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  {{- if and (eq .Values.master.service.type "LoadBalancer") (not (empty .Values.master.service.loadBalancerSourceRanges)) }}
+  loadBalancerSourceRanges: {{ .Values.master.service.loadBalancerSourceRanges }}
+  {{- end }}
+  {{- if and (eq .Values.master.service.type "LoadBalancer") (not (empty .Values.master.service.loadBalancerIP)) }}
+  loadBalancerIP: {{ .Values.master.service.loadBalancerIP }}
+  {{- end }}
+  ports:
+    - name: http
+      port: {{ .Values.master.service.ports.http }}
+      {{- if not (eq .Values.master.service.ports.http .Values.master.containerPorts.http) }}
+      targetPort: {{ .Values.master.containerPorts.http }}
+      {{- end }}
```

Comment:
There's a reason why headless service simplify this block: we're using `.Values.master.containerPorts.http` to set the headless service port (instead of `.Values.master.service.ports.http`).
Why? Because of the flags we're passing to `weed` to start Master server:
```yaml
            - -ip=$(POD_NAME).{{ printf "%s-headless" (include "seaweedfs.master.fullname" .) }}.$(NAMESPACE).svc.{{ .Values.clusterDomain }}
            - -port={{ .Values.master.containerPorts.http }}
            - -peers={{ include "seaweedfs.master.servers" . }}
```

That renders like this with 3 replicas:

```yaml
            - -ip=$(POD_NAME).seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local
            - -port=9333
            - -peers=seaweedfs-master-0.seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local:9333,seaweedfs-master-1.seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local:9333,seaweedfs-master-2.seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local:9333
```

There's no problem when we set the same value to `master.containerPorts.http` and `master.service.ports.http`. However, if we set a different one (e.g. `master.service.ports.http=80`) the rendering would look like this:

```yaml
            - -ip=$(POD_NAME).seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local
            - -port=9333
            - -peers=seaweedfs-master-0.seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local:80,seaweedfs-master-1.seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local:80,seaweedfs-master-2.seaweedfs-master-headless.$(NAMESPACE).svc.cluster.local:80
```

And.. It should be fine but it seems there's some limitation in `weed` and when you start it throws the error below:

```
I0412 07:45:22.246909 master.go:269 current: seaweedfs-master-0.seaweedfs-master-headless.default.svc.cluster.local:9333 peers:seaweedfs-master-0.seaweedfs-master-headless.default.svc.cluster.local:80,seaweedfs-master-1.seaweedfs-master-headless.default.svc.cluster.local:80,seaweedfs-master-2.seaweedfs-master-headless.default.svc.cluster.local:80
F0412 07:45:22.246921 master.go:285 Only odd number of masters are supported: [seaweedfs-master-0.seaweedfs-master-headless.default.svc.cluster.local:80 seaweedfs-master-1.seaweedfs-master-headless.default.svc.cluster.local:80 seaweedfs-master-2.seaweedfs-master-headless.default.svc.cluster.local:80 seaweedfs-master-0.seaweedfs-master-headless.default.svc.cluster.local:9333]
```

It is not recognizing itself in the peer list due to the port change. That's why I decided to provide support for different ports exclusively for the Service and keep the Headless Service using the "container port".

Reply:
done at https://github.com/bitnami/charts/pull/24944/commits/f95b62e4cfec5af87c908ab293d2f105a2d05dff